crucible-llvm/src/Lang/Crucible/LLVM/Intrinsics/Libcxx.hs

```diff
@@ -130,17 +128,16 @@ panic_ from decl args ret =
 mkOverride :: (IsSymInterface sym, HasPtrWidth wptr, wptr ~ ArchWidth arch)
            => String -- ^ Name (for 'panic')
            -> [String] -- ^ Substrings for name filtering
-           -> (LLVMHandleInfo -> Maybe (SomeLLVMOverride p sym arch))
+           -> (forall args ret. L.Declare -> CtxRepr args -> TypeRepr ret -> Maybe (SomeLLVMOverride p sym arch))
            -> (L.Symbol -> ABI.DecodedName -> Bool)
            -> SomeCPPOverride p sym arch
-mkOverride name substrings ov filt =
+mkOverride _name substrings ov filt =
   SomeCPPOverride substrings $ \requestedDecl decodedName llvmctx ->
+    let ?lc = llvmctx^.llvmTypeCtx in
     matchSymbolName filt requestedDecl decodedName $
-      case (Map.lookup (L.decName requestedDecl) (llvmctx ^. symbolMap)) of
-        Nothing         -> panic name [ "No function handle for symbol:"
-                                      , show (L.decName requestedDecl)
-                                      ]
-        Just handleInfo -> ov handleInfo
+      do decl' <- either fail return $ liftDeclare requestedDecl
```

Comment:
Is `fail` the correct error behavior here? Is failure a bug in Crucible? Is it recoverable? Does it indicate a malformed LLVM module?

Reply:
I think it probably indicates a malformed LLVM module... I don't think it's recoverable, but I don't think it's a `panic` either (i.e., I don't think it corresponds to the failure of an intended invariant).

Reply:
Can we make a helper function `malformedLLVMModule :: String -> a` which makes that more explicit? It'd be great to spread use of that more through the codebase in a follow-up issue.

Reply:
Yeah, that's probably a good idea.